runtime/doc/lua.txt

```diff
@@ -1996,6 +1996,20 @@ vim.list_slice({list}, {start}, {finish})                   *vim.list_slice()*
     Return: ~
         (list) Copy of table sliced from start to finish (inclusive)
 
+vim.memoize({fn}, {hash_fn})                                   *vim.memoize()*
+    Memoize a function {fn} using {hash_fn} to hash the arguments. Internally
+    uses a weak table to cache the results of {fn}.
+
+    Parameters: ~
+      • {fn}       (function) Function to memoize.
+      • {hash_fn}  integer|'concat'|function|nil Hash function to create a
+                   hash to use as a key to store results. When an integer,
+                   refers to the argument index to use as the hash. When 'concat `, the hash is determined by string concatenating all the
+                   arguments. Defaults to 1.
```

Comment:
We could consider removing the default and making this parameter required, so that no one mistakenly accepts the default when they shouldn't (the percentage of cases where "1" is correct may not be high enough to justify it)